lib/services/swagger-api-service.js

```diff
@@ -163,15 +163,16 @@ function swaggerFactory(
             }
 
             return render.then(function(data) {
-                if (!data || typeof(data) !== 'string') {
-                    var err = new Errors.TemplateRenderError(
-                        viewName + ' could not be rendered.'
-                    );
-                    err.status = 500;
-                    throw(err);
+                // Rendered data should be JSON string.
+                try {
+                    JSON.parse(data);
+                } catch(err) {
+                    throw new Errors.ViewRenderError(err.message);
                 }
-                // Remove whitespace
                 return data.replace(/\s/g, '');
+            })
+            .catch(function(err) {
```

Comment:
Strictly speaking, you're right.  The reason I added it is to convert an AssertionError to a ViewRenderError so that the status property is set properly for the swagger render fitting.  Since both AssertionError and ViewRenderError result in 500 status, it ultimately doesn't matter but there could be problems if other unexpected errors are thrown or if we decide that we want to change the status code for ViewRenderError.